lib/access_control_model_tag.rb

```diff
@@ -20,4 +20,86 @@ def applied_dataset
   def add_member(member_id)
     applied_dataset.insert(:tag_id => id, applied_column => member_id)
   end
+
+  def member_ids
+    applied_dataset.where(tag_id: id).select_map(applied_column)
+  end
+
+  def add_members(member_ids)
+    applied_dataset.import([:tag_id, applied_column], member_ids.map { [id, _1] })
+  end
+
+  def remove_members(member_ids)
+    applied_dataset.where(:tag_id => id, applied_column => member_ids).delete
```

Comment:
Should we use `destroy` instead of `delete`? Otherwise, the data won't be archived to the ArchivedRecord table, which we use for auditing operations.

Reply:
You cannot currently use `destroy`, because there is no model for the `applied_*_tag` tables.

If we want to copy deleted records for the `applied_*_tag` tables to the archive table, my recommendation would be to use database triggers and not model hooks.  That is faster and more robust.  `Dataset#destroy` is quite inefficient, because it retrieves all rows, and deletes each row in a separate query, running hooks separately for each row.  Please let me know if you would like me to work on a migration for that.

Reply:
Database triggers seem like a good idea to me. We may also get feedback from other team members too. What's your opinion on this? @byucesoy @fdr

Reply:
I pushed a commit with a migration that uses database triggers on `applied_{subject,action,object}_tag` to insert into archived_records, as well as a spec that shows that it works.